Source/Embeddings.Processing/EmbeddingProcessor.cs

```diff
@@ -163,8 +167,14 @@ async Task<Try> Loop()
                 return Try.Succeeded();
             }
             _logger.CommittingTransitionEvents(_embedding, key, uncommittedEvents);
-            var committedEvents = await _eventStore.CommitAggregateEvents(uncommittedEvents.Result, executionContext, cancellationToken).ConfigureAwait(false);
-            await replaceOrRemoveEmbedding(committedEvents[^1].AggregateRootVersion + 1).ConfigureAwait(false);
+            var response = await _eventStore.CommitAggregateEvents(uncommittedEvents.Result.ToCommitRequest(executionContext), cancellationToken).ConfigureAwait(false);
+            if (response.Failure != default)
+            {
+                return new FailedToCommitEmbeddingEvents(response.Failure.ToFailure());
+            }
+
+            var committedEvents = response.Events.ToCommittedEvents();
+            await replaceOrRemoveEmbedding(committedEvents.Last().AggregateRootVersion + 1).ConfigureAwait(false);
```

Comment:
Does this make sense, @jakhog ?

Reply:
This looks to me like the same behaviour as before - so from a quick look, it looks correct :) Aren't there specs that check this?

Reply:
There are a few :)